unstable test case TestIssue29282

```
[2022-03-15T06:27:28.046Z] --- FAIL: TestIssue29282 (0.61s)
[2022-03-15T06:27:28.046Z]     testkit.go:108: 
[2022-03-15T06:27:28.046Z]         	Error Trace:	testkit.go:108
[2022-03-15T06:27:28.046Z]         	            				db_integration_test.go:3679
[2022-03-15T06:27:28.047Z]         	            				asm_amd64.s:1371
[2022-03-15T06:27:28.047Z]         	Error:      	Received unexpected error:
[2022-03-15T06:27:28.047Z]         	            	context canceled
[2022-03-15T06:27:28.047Z]         	            	github.com/pingcap/errors.AddStack
[2022-03-15T06:27:28.047Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
[2022-03-15T06:27:28.047Z]         	            	github.com/pingcap/errors.Trace
[2022-03-15T06:27:28.047Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15
[2022-03-15T06:27:28.047Z]         	            	github.com/pingcap/tidb/store/driver/error.ToTiDBErr
[2022-03-15T06:27:28.047Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_check/go/src/github.com/pingcap/tidb/store/driver/error/error.go:169
[2022-03-15T06:27:28.047Z]         	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTaskOnce
[2022-03-15T06:27:28.047Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_check/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:762
[2022-03-15T06:27:28.047Z]         	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask
[2022-03-15T06:27:28.047Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_check/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:673
[2022-03-15T06:27:28.047Z]         	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).run
[2022-03-15T06:27:28.047Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_check/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:414
[2022-03-15T06:27:28.047Z]         	            	runtime.goexit
[2022-03-15T06:27:28.047Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1371
[2022-03-15T06:27:28.047Z]         	Test:       	TestIssue29282
[2022-03-15T06:27:28.047Z]         	Messages:   	sql:select * from issue29828_t where id = 1 for update;, args:[]
```

_Originally posted by @kennytm in https://github.com/pingcap/tidb/issues/32687#issuecomment-1067683955_

--- a/ddl/db_integration_test.go
+++ b/ddl/db_integration_test.go
@@ -3757,6 +3757,13 @@ func TestIssue29282(t *testing.T) {
 		// Unexpected, test fail.
 		t.Fail()
 	}
+
+	// Wait the background query rollback
+	select {
+	case <-time.After(100 * time.Millisecond):
+		t.Fail()
+	case <-ch:
+	}
 }
 
 // See https://github.com/pingcap/tidb/issues/29327

